docs/testing.md

```diff
@@ -0,0 +1,195 @@
+We enjoy improving Verbs by providing easy, readable testing affordances.
+
+## `Verbs::Commit()`
+
+When testing verbs events, you'll need to call [commit](/docs/reference/events#content-committing) manually.
+
+You may continue manually adding `Verbs::commit()` after each `Event::fire()` method; however, we've created `Verbs::commitImmediately` to issue a blanket commit on all events you fire in tests.
+
+```php
+beforeEach(function () {
+    Verbs::commitImmediately();
+});
+```
+
+You may also implement the `CommitsImmediately` interface directly on an Event.
+(more about this in [`VerbsStatesInitialized`](testing#content-verbsstateinitialized))
+
+### Assertions
+
+The following Test `assert()` methods are available to thoroughly check your committing granularly.
+
+```php
+Verbs::assertNothingCommitted();
+Verbs::assertCommitted(...);
+Verbs::assertNotCommitted(...);
+```
+
+## State Factories
+
+In tests, you may find yourself needing to fire and commit several events in order to bring your State to the point where it actually needs testing.
+
+The `ExampleState::factory()` method allows you to bypass manually building up the State, functioning similarly to `EloquentModel::factory()->state(['data'=>'data'])->create()`
+
+This allows you to call:
+
+```php
+BankAccountState::factory()->create(
+    data: ['balance' => 1337]
+    id: $bank_account_id
+);
+
+// Or, using `id()` syntax:
+
+BankAccountState::factory()
+    ->id($bank_account_id)
+    ->create(
+        data: ['balance' => 1337]
+    );
+```
+
+- If you accidentally pass an ID into both `id()` and `create()`, `create()` takes precedence.
+
+Or, in the case of a [singleton state](/docs/reference/states#content-singleton-states):
+
+```php
+ChurnState::factory()->create(['churn' => 40]);
+```
+
+Next, we'll get into how these factories work, and continue after with some [Verbs factory methods](testing#content-factory-methods) you may already be familiar with from Eloquent factories.
+
+### `VerbsStateInitialized`
+
+Under the hood, these methods will fire (and immediately commit) a new `VerbsStateInitialized` event, which will fire onto the given state, identified by the id argument (if id is null, we assume it is a singleton) and return a copy of that state.
+
+This is primarily designed for booting up states for testing, but it may also be useful in migrating non event-sourced codebases to Verbs, when there is a need to initiate a state for legacy data.
```

Comment:
I don't think we should suggest using this in prod. It's better to create your own "migrated from legacy" event in that case.